test/utils/Checkpoints.test.js

```diff
@@ -4,71 +4,144 @@ const { expect } = require('chai');
+      });
+
+      it('multiple checkpoints in the same block', async function () {
+        const lengthBefore = await this.checkpoint.length();
+        await batchInBlock([
+          () => this.checkpoint.push(8, { gas: 100000 }),
+          () => this.checkpoint.push(9, { gas: 100000 }),
+          () => this.checkpoint.push(10, { gas: 100000 }),
+        ]);
+        const lengthAfter = await this.checkpoint.length();
+
+        expect(lengthAfter.toNumber()).to.be.equal(lengthBefore.toNumber() + 1);
+        expect(await this.checkpoint.latest()).to.be.bignumber.equal('10');
+      });
     });
   });
 
-  describe('with checkpoints', function () {
-    beforeEach('pushing checkpoints', async function () {
-      this.tx1 = await this.checkpoint.push(1);
-      this.tx2 = await this.checkpoint.push(2);
-      await time.advanceBlock();
-      this.tx3 = await this.checkpoint.push(3);
-      await time.advanceBlock();
-      await time.advanceBlock();
-    });
+  for (const length of [160, 224]) {
+    describe(`Checkpoint${length}[]`, function () {
+      beforeEach(async function () {
+        this.contract = await artifacts.require(`Checkpoints${length}Mock`).new();
+      });
 
-    it('returns latest value', async function () {
-      expect(await this.checkpoint.latest()).to.be.bignumber.equal('3');
-    });
+      describe('without checkpoints', function () {
+        it('returns zero as latest value', async function () {
+          expect(await this.contract.latest()).to.be.bignumber.equal('0');
+        });
 
-    it('returns past values', async function () {
-      expect(await this.checkpoint.getAtBlock(this.tx1.receipt.blockNumber - 1)).to.be.bignumber.equal('0');
-      expect(await this.checkpoint.getAtBlock(this.tx1.receipt.blockNumber)).to.be.bignumber.equal('1');
-      expect(await this.checkpoint.getAtBlock(this.tx2.receipt.blockNumber)).to.be.bignumber.equal('2');
-      // Block with no new checkpoints
-      expect(await this.checkpoint.getAtBlock(this.tx2.receipt.blockNumber + 1)).to.be.bignumber.equal('2');
-      expect(await this.checkpoint.getAtBlock(this.tx3.receipt.blockNumber)).to.be.bignumber.equal('3');
-      expect(await this.checkpoint.getAtBlock(this.tx3.receipt.blockNumber + 1)).to.be.bignumber.equal('3');
-    });
+        it('lookup returns 0', async function () {
+          expect(await this.contract.lowerLookup(0)).to.be.bignumber.equal('0');
+          expect(await this.contract.upperLookup(0)).to.be.bignumber.equal('0');
+          expect(await this.contract.upperLookupRecent(0)).to.be.bignumber.equal('0');
+        });
+      });
 
-    it('reverts if block number >= current block', async function () {
-      await expectRevert(
-        this.checkpoint.getAtBlock(await web3.eth.getBlockNumber()),
-        'Checkpoints: block not yet mined',
-      );
+      describe('with checkpoints', function () {
+        beforeEach('pushing checkpoints', async function () {
+          this.checkpoints = [
```

Comment:
This is scarily simple for "coverage". I'd change it so the values are mostly not equal to the keys, and not in increasing order.